dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -125,15 +125,26 @@ case class BinaryOutcomeDLCWithSelf(
                                Vector(fundingLocalPubKey, fundingRemotePubKey))
   }
 
-  /** Subtracts the estimated fee from the last output's value */
+  /** Subtracts the estimated fee by removing from each output evenly */
   private def subtractFeeAndSign(
       txBuilder: BitcoinTxBuilder): Future[Transaction] = {
     txBuilder.unsignedTx.flatMap { tx =>
       val fee = feeRate.calc(tx)
-      val output = txBuilder.destinations.last
-      val newOutput = TransactionOutput(output.value - fee, output.scriptPubKey)
+
+      val outputs = txBuilder.destinations
+
+      val feePerOutput = Satoshis(Int64(fee.satoshis.toLong / outputs.length))
+      val feeRemainder = Satoshis(Int64(fee.satoshis.toLong % outputs.length))
+
+      val newOutputsWithoutRemainder = outputs.map(output =>
+        TransactionOutput(output.value - feePerOutput, output.scriptPubKey))
+      val lastOutput = newOutputsWithoutRemainder.last
+      val newLastOutput = TransactionOutput(lastOutput.value - feeRemainder,
+                                            lastOutput.scriptPubKey)
+      val newOutputs = newOutputsWithoutRemainder.dropRight(1).:+(newLastOutput)
+
       val newBuilder =
-        BitcoinTxBuilder(txBuilder.destinations.dropRight(1).:+(newOutput),
+        BitcoinTxBuilder(newOutputs,
```

Comment:
Can we get a test case on this that this is actually right? I think that `TxBuilder` will try to re-subtract fees again when we call `.sign()`?

Reply:
Added a test that checks that building fails before subtraction and works (as expected, with evenly distributed fees among outputs) after.

To clarify, `TxBuilder` never subtracts fees from anything, it expects `sum(inputValues) > sum(outputValues)` and throws the difference minus fees into a change SPK. That is why this is necessary and why it works as expected.